src/main/java/com/iota/iri/service/validation/impl/TransactionSolidifierImpl.java

```diff
@@ -0,0 +1,276 @@
+        Hash hash;
+        while (!Thread.currentThread().isInterrupted()) {
+            if((hash = transactionsToSolidify.poll()) != null) {
+                try {
+                    checkSolidity(hash);
+                } catch (Exception e) {
+                    log.info(e.getMessage());
+                }
+            }
+        }
+    }
+
+    /**
+     *{@inheritDoc}
+     */
+    @Override
+    public boolean checkSolidity(Hash hash) throws Exception {
+        return checkSolidity(hash, Integer.MAX_VALUE);
+    }
+
+    /**
+     *{@inheritDoc}
+     */
+   @Override
+    public boolean checkSolidity(Hash hash, int maxProcessedTransactions) throws Exception {
+        if(fromHash(tangle, hash).isSolid()) {
+            return true;
+        }
+        LinkedHashSet<Hash> analyzedHashes = new LinkedHashSet<>(snapshotProvider.getInitialSnapshot().getSolidEntryPoints().keySet());
+        if(maxProcessedTransactions != Integer.MAX_VALUE) {
+            maxProcessedTransactions += analyzedHashes.size();
+        }
+        boolean solid = true;
+        final Queue<Hash> nonAnalyzedTransactions = new LinkedList<>(Collections.singleton(hash));
+        Hash hashPointer;
+        while ((hashPointer = nonAnalyzedTransactions.poll()) != null) {
+            if (!analyzedHashes.add(hashPointer)) {
+                continue;
+            }
+
+            if (analyzedHashes.size() >= maxProcessedTransactions) {
+                return false;
+            }
+
+            TransactionViewModel transaction = fromHash(tangle, hashPointer);
+            if (isUnsolidWithoutEntryPoint(transaction, hashPointer)) {
+                if (transaction.getType() == PREFILLED_SLOT) {
+                    solid = false;
+                    checkRequester(hashPointer);
+                } else {
+                    nonAnalyzedTransactions.offer(transaction.getTrunkTransactionHash());
+                    nonAnalyzedTransactions.offer(transaction.getBranchTransactionHash());
+                }
+            }
+        }
+        if (solid) {
+          updateTransactions(analyzedHashes);
+        }
+        analyzedHashes.clear();
+        return  solid;
+    }
+
+
+    /**
+     * Check if a transaction is present in the {@link #transactionRequester}, if not, it is added.
+     * @param hashPointer   The hash of the transaction to request
+     */
+    private void checkRequester(Hash hashPointer){
+        if (!transactionRequester.isTransactionRequested(hashPointer)) {
+            transactionRequester.requestTransaction(hashPointer);
+        }
+    }
+
+    /**
+     * Iterate through analyzed hashes and place them in the {@link #transactionsToBroadcast} queue
+     * @param hashes    Analyzed hashes from the {@link #checkSolidity(Hash)} call
+     */
+    private void updateTransactions(Set<Hash> hashes) {
+        hashes.forEach(hash -> {
+            try {
+                TransactionViewModel tvm = TransactionViewModel.fromHash(tangle, hash);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'TransactionViewModel.fromHash' due to existing static import 'com.iota.iri.controllers.TransactionViewModel.fromHash'](https://app.codacy.com/app/GalRogozinski/iri/pullRequest?prid=4339340)